modules/nlu/src/backend/api.ts

```diff
@@ -118,6 +118,64 @@ export const registerRouter = async (bp: typeof sdk, app: NLUApplication) => {
       return mapError({ botId, lang, error }, res)
     }
   })
+
+  addTrainingServiceRoutes(router, app, mapError)
+}
+
+const trainingStatusSchema = Joi.object().keys({
+  status: Joi.string()
+    .valid(['idle', 'done', 'needs-training', 'training-pending', 'training', 'canceled', 'errored'])
+    .required(),
+  progress: Joi.number()
+    .default(0)
+    .min(0)
+    .max(1)
+    .required()
+})
+
+const addTrainingServiceRoutes = (
+  router: sdk.http.RouterExtension,
+  app: NLUApplication,
+  errorMapper: (err: { botId: string; lang: string; error: Error }, res: Response) => Response
+) => {
+  const repo = app.repository
+  const SUBROUTE_NAME = 'trainrepo'
+
+  router.get(`/${SUBROUTE_NAME}/trainings`, async (req, res) => {
+    const trainings = await repo.getAll()
+    res.status(200).json(trainings)
+  })
+
+  router.use(`/${SUBROUTE_NAME}/trainings/:botId/:lang`, async (req, res, next) => {
+    const { botId, lang } = req.params
+    try {
+      const training = await repo.get({ botId, language: lang })
+      if (!training) {
+        throw new Error('Training not found')
+      }
+      res.locals.training = training
```

Comment:
what's this? I don't know this feature

Reply:
It's the context store for the request, the middleware will always be executed first for all the matching routes, if we try to do any operations on an inexistant training, the middleware will handle the check without having to duplicate code. Storing the result of the query in the request context store avoids querying twice